libraries/autoload.rb

```diff
@@ -1,3 +1,7 @@
+unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
+  raise "The compat_resources cookbook does not support chef versions older than Chef 12.0.0"
```

Comment:
If you're writing resources that require compat_resource and you no-op this cookbook, then the resources will fail to parse the first time you encounter something like the 'action' keyword in the resources DSL (which is a 12.5 extension).  i don't see how there's any way to make this work with chef 11.x, it fundamentally backports 12.5 features to 12.0-4 chef-clients in order to write resources using 12.5 syntax.